substrate-archive-backend/src/database.rs

```diff
@@ -56,7 +57,16 @@ impl ReadOnlyDatabase {
         })
     }
 
-    pub fn get(&self, col: ColumnId, key: &[u8]) -> Option<Vec<u8>> {
+    pub fn catch_up_count(&self) -> Option<usize> {
+        if !self.track_catchups {
+            log::warn!("catchup tracking is not enabled");
+            None
+        } else {
+            Some(self.catch_counter.fetch_add(0, Ordering::Relaxed))
```

Comment:
not really introduced in this PR but `self.catch_counter.fetch_add(0, Ordering::Relaxed)` looks strange, it's better to use `load` if we just want read the value.